lib/internal/abort_controller.js

```diff
@@ -316,6 +336,26 @@ class AbortController {
   }
 }
 
+/**
+ * Enables the AbortSignal to be transferable using structuredClone/postMessage.
+ * @param {AbortSignal} signal
+ * @returns {AbortSignal}
+ */
+function transferableAbortSignal(signal) {
+  if (signal?.[kAborted] === undefined)
+    throw new ERR_INVALID_ARG_TYPE('signal', 'AbortSignal', signal);
+  return makeTransferable(signal);
+}
+
+/**
+ * Creates an AbortController with a transferable AbortSignal
+ */
+function transferableAbortController() {
```

Comment:
I think the private field still works if you make this a static function of AbortController?

Reply:
We should definitely not add non-standard static functions to the standard API

Reply:
Oh, right, I forgot that this is part of the a standard - then how about making it a static symbol function and exporting it to util using that symbol?

Reply:
We will give that a try.

Reply:
i always enjoyed this pattern:
```
class Foo {
  #bar;

  static myUtilFunction() {
     // do something with #bar
  }
}

const { myUtilFunction } = Foo;
delete Foo.myUtilFunction;

Reply:
Cam you definethat function in a static block?

```js
let transferableAbortController;

class AbortController {
  #signal; 

  static {
    transferableAbortController = ...
  }
}
```

Reply:
> EDIT: no, that still wouldn't work because you cannot install private fields on objects that previously dis not have them (without using some complex tricks).

Oh, yes, I didn't think of the private checks.

With this specific case though, single all the new option does is to wrap the new signal with a call of `lazyMakeTransferable()`, I wonder if this should already work:

```js
function transferableAbortController() {
  const controller = new AbortController();
  controller[kSetSignal](lazyMakeTransferable(controller.signal));
  return controller;
}
```

Where `kSetSignal` is a setter of `#signal` that's only available to internals. Similarly `ClonedAbortSignal()` only needs to call `lazyMakeTransferable()` on a new signal before returning it.

Reply:
I'm not convinced that is any improvement here. For the first case, it ends up creating an extra AbortSignal that is immediately discarded, and in both cases we still end up with a hidden symbol on the AbortController. The use of the private property vs a hidden symbol really isn't that compelling of an argument so it just feels like needless bike shedding a bit.

Reply:
> For the first case, it ends up creating an extra AbortSignal that is immediately discarded

From what I can tell no new signals should be constructed? I think the current patch does exactly the same thing to that signal, just in different order? Here it constructs the signal, calls makeTransferable() on it (which constructs a JSTransferrable and copies things over), and returns the copied signal, discarding the original one. The suggestion just adds a reference to the original signal in the default constructor, and later that reference is replaced with a reference to the copied signal. The cost of the intermediate reference is negligible compared to the cost of the ReflectConstruct call added here. Even if the AbortController isn’t using private fields, I think the suggestion can still be applied to get rid of the additional ReflectConstruct - it just happens to make private fields viable at the same time. This way it’s not necessary to add back the type checks in this patch either and V8 would handle it slightly more efficiently (the missing type checks are currently failing the tests below BTW).

Anyway, my suggestion isn’t blocking. I thought I was pinged for suggestions to avoid reverting the previous patch, and that’s what I tried to do, sorry if I caused any troubles for anyone involved here. I am okay with this patch as it is, minus that type checks should be added to make the tests pass.

Reply:
> From what I can tell no new signals should be constructed? 

Heh, I misremembered my own code ;-) ... @flakey5 updated the PR to address this concern.